### PR TITLE
Fix deploy regex in .circleci/config.yml

### PR DESCRIPTION
**Pull request by arcondello**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -345,37 +345,21 @@ workflows:
     jobs:
       - build-linux:
           <<: *build
-          filters:
+          filters: &deploy-filters
             tags:
-              only: /^[0-9]+(\.[0-9]+)*((\.dev|rc)([0-9]+)?)?$/
+              only: /^[0-9]+(\.[0-9]+)*(\.(dev|rc|post)([0-9]+)?)?$/
```
Comment (randomir) [inline .circleci/config.yml]: If I'm reading [PEP-440](https://www.python.org/dev/peps/pep-0440/) correctly, normalized forms are:
```
rc
.post
.dev
```

All three start at zero, i.e. `rc == rc0`, etc.

In addition to `.`, post and dev can start with `-` and `_`, but they are normalized to `.`.

Comment (arcondello) [reply .circleci/config.yml]: Oh interesting. I thought that was an odd bug, should have checked it myself. Thanks! Will fix.

Comment (arcondello) [reply .circleci/config.yml]: Should be fixed.

**Review by randomir — APPROVED** [review]
LGTM

**Pull request merged**